### PR TITLE
[Azure] Integration test for Attach resource

### LINKED ISSUE #405: [Azure] Add integration test for AttachResource
- [x] Add integration test for attach resource function in plugin

_Originally posted by @J-467 in https://github.com/paraglider-project/paraglider/issues/170#issuecomment-2161210207_

### PR DESCRIPTION
**Pull request by J-467**
- Integration test for Attach Resource
- Closes #405

### CODE REVIEW

File: pkg/azure/plugin_integration_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func createVM(ctx context.Context, server *azurePluginServer, subscriptionId string, resourceGroupName string, namespace string, location string, name string) (*paragliderpb.CreateResourceResponse, error) {
+func createParagliderVM(ctx context.Context, server *azurePluginServer, subscriptionId string, resourceGroupName string, namespace string, location string, name string) (*paragliderpb.CreateResourceResponse, error) {
```
Comment (seankimkdy) [inline pkg/azure/plugin_integration_test.go:35]: Any reason this was renamed? If this clashes with another name, then a better name might be `createTestVM`. We don't have a concept known as a "Paraglider VM", so the method might be misleading.

Also, I prefer to keep names short as Go is already verbose.

File: pkg/azure/plugin_integration_test.go [hunk 2]
```diff
@@ -360,3 +360,168 @@ func TestMultipleRegionsIntraNamespace(t *testing.T) {
...
+		SubscriptionID:    subscriptionId,
+		ResourceGroupName: resourceGroupName,
+		ResourceName:      externalVmName,
+	}
+	externalAddressSpace := "20.9.0.0/24"
+	azureHandler, err := azureServer.setupAzureHandler(resourceIdInfo, namespace)
+	require.NoError(t, err)
+
+	// Create Non-Paraglider Vnet
+	externalVnetParams := CreateVirtualNetworkParameters(ctx, vmLocation, externalVnetName, externalAddressSpace)
```
Comment (seankimkdy) [inline pkg/azure/plugin_integration_test.go:418]: This probably should have been addressed in a previous PR, but could we rename this to `GetVirtualNetworkParameters`? Create makes it sound like it's going to create the VNET.

And also, we shouldn't export this method unless there is a major reason to use this in other packages.

**Review by seankimkdy — CHANGES_REQUESTED** [review]

File: pkg/azure/utils.go [hunk 3]
```diff
@@ -529,3 +529,28 @@ func DoesVnetOverlapWithParaglider(ctx context.Context, handler *AzureSDKHandler
 
 	return false, nil
 }
+
+// CreateVirtualNetworkParameters creates and returns an instance of armnetwork.VirtualNetwork
+// with the specified parameters.
+//
+// Subnet address space is the same as the Vnet address space.
+func CreateVirtualNetworkParameters(ctx context.Context, location string, vnetName string, addressSpace string) armnetwork.VirtualNetwork {
```
Comment (seankimkdy) [inline pkg/azure/utils.go:537]: Could we rename this to GetVirtualNetworkParameters? Create makes it sound like it's going to create the VNET.

And also, we shouldn't export this method unless there is a major reason to use this in other packages.

**Review by seankimkdy — CHANGES_REQUESTED** [review]

File: pkg/azure/plugin_integration_test.go (hunk 1 above)
Comment (J-467) [reply pkg/azure/plugin_integration_test.go:35]: I was initially thinking of creating a function that returns a VM not created within Paraglider's vnet. However, I decided against that, so I would rename to as before

File: pkg/azure/utils.go (hunk 3 above)
Comment (J-467) [reply pkg/azure/utils.go:537]: Alright

File: pkg/azure/plugin_integration_test.go [hunk 4]
```diff
@@ -410,12 +412,28 @@ func TestAttachResourceIntegration(t *testing.T) {
...
+
+	// Find unused address spaces for external address
+	conn, err := grpc.NewClient(azureServer.orchestratorServerAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		utils.Log.Printf("Azure Integration test: Could not dial the orchestrator")
+		return
+	}
+	defer conn.Close()
+	client := paragliderpb.NewControllerClient(conn)
+	reqAddressSpaces := make([]int32, 2)
```
Comment (seankimkdy) [inline pkg/azure/plugin_integration_test.go:424]: Why length of 2? I thought we only need one external address space, in which we can actually pass `nil`.

Comment (J-467) [reply pkg/azure/plugin_integration_test.go:424]: You're right. We do need only 1. I had 2 for debugging purposes. Fixed this now. Thanks Sean

**Review by smcclure20 — APPROVED** [review]
LGTM! Thanks!

**Review by seankimkdy — APPROVED** [review]

**Pull request merged**